### PR TITLE
Add clipboard functionality for Allegro

### PR DESCRIPTION
**Pull request by Grant1219**
As of Allegro 5.1.12 there has been support for getting and setting clipboard text.

http://liballeg.org/a5docs/trunk/display.html#clipboard

### CODE REVIEW

File: source/platform/platforms/Allegro5Platform.cpp [hunk 1]
```diff
@@ -50,13 +50,22 @@ void Gwk::Platform::SetCursor(unsigned char iCursor)
 
 Gwk::String Gwk::Platform::GetClipboardText()
 {
-    return gs_ClipboardEmulator;
+    if (al_clipboard_has_text(g_display) {
+        char* clip = al_get_clipboard_text(g_display);
+
+        if (clip != NULL) {
```
Comment (billyquith) [inline source/platform/platforms/Allegro5Platform.cpp]: This project is C++11 so we can use `nullptr`.

File: source/platform/platforms/Allegro5Platform.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@
 static Gwk::Input::Allegro      g_gworkInput;
 static ALLEGRO_EVENT_QUEUE*     g_event_queue = nullptr;
 static ALLEGRO_DISPLAY*         g_display = nullptr;
-static Gwk::String              gs_ClipboardEmulator;
+static Gwk::String              gs_clipboard;
```
Comment (billyquith) [inline source/platform/platforms/Allegro5Platform.cpp]: If Allegro is managing the clipboard state I don't think we need to keep this here anymore.

**Comment by Grant1219** [comment]
Okay I removed the static variable, and also reformatted my code to match your style a bit better.

**Comment by billyquith** [comment]
Thanks.

**Pull request merged**